Hometasks1

pkay-14 opened the pull request.

MrCartoon commented on hometask1/hometask1-task8.rb:
```diff
@@ -0,0 +1,6 @@
+a = [23,4,56,1,9,4,76,2]
```
Change input array to ARGV(you already did it in another script).

MrCartoon commented on hometask1/hometask1-task1.rb:
```diff
@@ -0,0 +1,2 @@
+t = Time.now.getlocal()
```
This script doesn't do what was requested in the task.

MrCartoon commented on hometask1/hometask1-task5.rb:
```diff
@@ -0,0 +1,15 @@
+# path = File.expand_path(File.dirname(__FILE__))
+# path << filename
+
+b = get_line_number("#{__dir__}/#{filename}", line)
```
Here also `__dir__` isn't required if you want to be able to specify the dynamic file's location.

MrCartoon commented on hometask1/hometask1-task5.rb:
```diff
@@ -0,0 +1,15 @@
+# path << filename
+
+b = get_line_number("#{__dir__}/#{filename}", line)
+puts "line number of '#{line}' where first encountered in '#{filename}' is #{b}"
```
Add another output if requested line wasn't found, cause if you run `ruby hometask1-task5.rb mytext.txt "non-exiting text"` you will get `line number of 'non-exiting text' where first encountered in 'mytext1.txt' is`.

pkay-14 commented:
My apologies i saw this email late. I have corrected and changed most of the request you asked. I only
have two issues. It’s task 1 and some part of task 2.

Regarding task1 i’ve made more research but i still don’t get how exactly i should write the script.

About task2, i made few changes but i’m not able to use the colour from the argv and implement it in the script. I’ve searched the internet but not finding a helpful solution yet. Can you please suggest how.

I will create another pull request to see what i’ve changed so far.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

From: Sviat Safronov<mailto:notifications@github.com>
Sent: 14 November 2019 23:20
To: pkay-14/ma-ruby<mailto:ma-ruby@noreply.github.com>
Cc: Roland<mailto:rquaye15@hotmail.co.uk>; Author<mailto:author@noreply.github.com>
Subject: Re: [pkay-14/ma-ruby] Hometasks1 (#1)


@MrCartoon requested changes on this pull request.


In hometask1/hometask1-task2.rb<https://github.com/pkay-14/ma-ruby/pull/1#discussion_r346537407>:

> +  path = File.expand_path(File.dirname(__FILE__))

+  path << filename

+

+  file = File.open(path, 'r')

Why don't you want to use just File.open(filename, 'r')?


In hometask1/hometask1-task2.rb<https://github.com/pkay-14/ma-ruby/pull/1#discussion_r346538625>:

> +require 'colorize'

+

+colour = ARGV[0]

+filename = ARGV[1]

+

+ #filename.each{|filename| IO.readlines(filename).each{|line|puts line}}

+

+ #file = '\mytext.txt'

+

+

+  path = File.expand_path(File.dirname(__FILE__))

+  path << filename

+

+  file = File.open(path, 'r')

+  file.each do|line|

+    puts line.colorize(:colour)

It doesn't seem to be colorized. Check what's wrong.
Also, lines printed with an extra newline, it shouldn't be.


In hometask1/hometask1-task3.rb<https://github.com/pkay-14/ma-ruby/pull/1#discussion_r346540032>:

> +arr1 = ['boy', 'girl', 'animal', 'tree']

+arr2 = ['book', 'building', 'boy', 'street', 'animal']

Please change the expected arrays to ARGV as ruby hometask1-task3.rb one,two,three two,three,four execution command.


In hometask1/hometask1-task3.rb<https://github.com/pkay-14/ma-ruby/pull/1#discussion_r346541066>:

> +#puts arr1 & arr2

+

+arr1.each do |x|

+  if (arr2.include?(x))

+    print "#{x} "

+  end

+end

Commented code is faster & better to understand.
You can use join method on the final array in order to format it.


In hometask1/hometask1-task8.rb<https://github.com/pkay-14/ma-ruby/pull/1#discussion_r346544525>:

> @@ -0,0 +1,6 @@

+a = [23,4,56,1,9,4,76,2]

Change input array to ARGV(you already did it in another script).


In hometask1/hometask1-task1.rb<https://github.com/pkay-14/ma-ruby/pull/1#discussion_r346545021>:

> @@ -0,0 +1,2 @@

+t = Time.now.getlocal()

This script doesn't do what was requested in the task.


In hometask1/hometask1-task5.rb<https://github.com/pkay-14/ma-ruby/pull/1#discussion_r346549048>:

> @@ -0,0 +1,15 @@

+filename = ARGV[0]

+line = ARGV[1]

+

+def get_line_number(file, word)

+  line_num = File.foreach(file).with_index do |line, index |

+    break index + 1 if line.include? word

+  end

+  return line_num unless line_num.nil?

+end

+

+# path = File.expand_path(File.dirname(__FILE__))

+# path << filename

+

+b = get_line_number("#{__dir__}/#{filename}", line)

Here also __dir__ isn't required if you want to be able to specify the dynamic file's location.


In hometask1/hometask1-task5.rb<https://github.com/pkay-14/ma-ruby/pull/1#discussion_r346550105>:

> @@ -0,0 +1,15 @@

+filename = ARGV[0]

+line = ARGV[1]

+

+def get_line_number(file, word)

+  line_num = File.foreach(file).with_index do |line, index |

+    break index + 1 if line.include? word

+  end

+  return line_num unless line_num.nil?

+end

+

+# path = File.expand_path(File.dirname(__FILE__))

+# path << filename

+

+b = get_line_number("#{__dir__}/#{filename}", line)

+puts "line number of '#{line}' where first encountered in '#{filename}' is #{b}"

Add another output if requested line wasn't found, cause if you run ruby hometask1-task5.rb mytext.txt "non-exiting text" you will get line number of 'non-exiting text' where first encountered in 'mytext1.txt' is.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/pkay-14/ma-ruby/pull/1?email_source=notifications&email_token=ALJSABVJSI5QRMLQBEYAP7DQTW6KXA5CNFSM4JL67YXKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCLUNK2A#pullrequestreview-317248872>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALJSABVRP22YBEDESZZOA3TQTW6KXANCNFSM4JL67YXA>.
